src/EventListener/JsonApiProblemExceptionListener.php

```diff
@@ -35,7 +35,8 @@ public function onKernelException(ExceptionEvent $event): void
         $request = $event->getRequest();
         if (
             false === mb_strpos($request->getRequestFormat(), 'json') &&
```

Comment:
Wouldn't it be better to do it like this:

```
            false === mb_strpos($request->getPreferredFormat(), 'json') &&
            false === mb_strpos((string) $request->getContentType(), 'json')
```

The preferred format first checks the request format (which can be set to json from the router).
If that one is not set, it looks at the first known Accept type.
So if that one is not json, we can still check the content-type to make a final decision.


What do you think?

Reply:
Just updated it to use `getPreferredFormat`.

Tests are not passing because of composer:2 from what I understand, I'll look into it.

Reply:
This package is using an old version of grumphp. Dont worry about it - will upgrade it in another issue soon :)